app/uk/gov/hmrc/agentaccesscontrol/service/ESAuthorisationService.scala

```diff
@@ -170,4 +174,25 @@ class ESAuthorisationService @Inject()(
           "accessGranted" -> result) ++ extraDetails
     )
 
+  def checkForRelationship(arn: Arn,
+                           maybeUserId: Option[String],
+                           taxIdentifier: TaxIdentifier)(
+      implicit ec: ExecutionContext,
+      hc: HeaderCarrier): Future[Boolean] =
+    for {
+      // Check whether Granular Permissions are enabled and opted-in for this agent.
+      // If so, when calling agent-client-relationships, we will check whether a relationship exists _for that specific agent user_.
+      // If Granular Permissions are not enabled or opted out, we only check whether a relationship exists with the agency.
```

Comment:
what about the case where the FF is on and the agent has opted in but the client is unassigned to any group? They should get access.

Reply:
(provided the agency is authorised for the client)

Reply:
That logic is in agent-client-relationship. In AAC we only check opted in/opted out

Reply:
ok thanks